.github/workflows/curl.yml

```diff
@@ -0,0 +1,41 @@
+name: curl Test
+
+on:
+  push:
+  workflow_call:
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@master
+    - name: autogen
+      run: ./autogen.sh
+    - name: configure
+      run: ./configure --enable-all
+    - name: make
+      run: make
+    - name: install
+      run: sudo make install
+    - uses: actions/checkout@master
+      with:
+        repository: curl/curl
+        path: curl
+    - name: Install test dependency
+      working-directory: ./curl
+      run: sudo apt-get install nghttp2
+    - name: curl buildconf
+      working-directory: ./curl
+      run: ./buildconf
+    - name: curl configure
+      working-directory: ./curl
+      run: ./configure --with-wolfssl
+    - name: curl make
+      working-directory: ./curl
+      run: make
```

Comment:
Probably a good idea to call `make -j` to get it building faster.

Reply:
Ah good suggestion. I'll add it when I'm doing another PR